File: backend/app/schemas/comment.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class CommentCreate(BaseModel):
    content: str
    post_id: UUID
    parent_id: UUID | None = None


class CommentUpdate(BaseModel):
    content: str


class CommentUserResponse(BaseModel):
    id: UUID
    first_name: str
    last_name: str
    username: str
    avatar: str | None = None

    model_config = ConfigDict(from_attributes=True)


class CommentResponse(BaseModel):
    id: UUID
    content: str
    user_id: UUID
    post_id: UUID
    parent_id: UUID | None
    created_at: datetime
    updated_at: datetime
    user: CommentUserResponse

    model_config = ConfigDict(from_attributes=True)

File: web/src/types/comment.ts
export type CommentUser = {
  id: string
  first_name: string
  last_name: string
  username: string
  avatar?: string | null
}

export type Comment = {
  id: string
  content: string
  user_id: string
  post_id: string
  parent_id?: string | null
  created_at: string
  updated_at: string
  user: CommentUser
  replies?: Comment[]
}

export type CommentCreatePayload = {
  content: string
  post_id: string
  parent_id?: string | null
}

export type CommentUpdatePayload = {
  content: string
}
